### PR TITLE
[Storage] Remove `loop` from file share

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This was discussed during API View presentation of STG84. Just a hanging param we should get rid of as Python as deprecated it.

### CODE REVIEW

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py [hunk 1]
```diff
@@ -144,13 +143,14 @@ def __init__(  # type: ignore
...
+            warnings.warn("``loop`` is no longer supported.", DeprecationWarning)
         super(ShareFileClient, self).__init__(
             account_url, share_name=share_name, file_path=file_path, snapshot=snapshot,
-            credential=credential, loop=loop, **kwargs
+            credential=credential, **kwargs
         )
-        self._client = AzureFileStorage(self.url, base_url=self.url, pipeline=self._pipeline, loop=loop)
+        self._client = AzureFileStorage(self.url, base_url=self.url, pipeline=self._pipeline)
         self._client._config.version = get_api_version(kwargs) # pylint: disable=protected-access
-        self._loop = loop
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:153]: If self._loop was never actually used, we probably don't need to be too verbose in the in the warnings, alternatively, if it _was_ actually being passed into to any asyncio APIs, we will need to raise a ValueError if this was passed in with >=Python 3.10.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py [hunk 2]
```diff
@@ -144,13 +143,14 @@ def __init__(  # type: ignore
         # type: (...) -> None
         kwargs["retry_policy"] = kwargs.get("retry_policy") or ExponentialRetry(**kwargs)
         loop = kwargs.pop('loop', None)
+        if loop:
+            warnings.warn("``loop`` is no longer supported.", DeprecationWarning)
```
Comment (annatisch) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:147]: We could more explicitly check for Python > 3.10. Also I think we can update the message to align with the [official Python statement](https://docs.python.org/3/whatsnew/3.10.html#whatsnew310-removed):
```python
if loop and sys.version_info >= (3, 8):
    warnings.warn("The 'loop' parameter was deprecated from asyncio's high-level APIs in Python 3.8 and is no longer supported.", DeprecationWarning)
```
Lastly as mentioned below - if we were actually using this anywhere - we'll need to upgrade from a warning to a ValueError.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:153]: Thanks for the feedback Anna 😄 I have gone back and confirmed that it is indeed redundant and not used anywhere else in the `azure-storage-file-share` package. With that being said, no `ValueError` needs to be raised, and in conjunction with your other comment, I will explicitly check and align with Python's official statement. 
Thanks!

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py (hunk 2 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:147]: See comment on other issue. Resolved 😄

**Pull request merged**